src/common/components/hosts/tableUtils.tsx

```diff
@@ -382,7 +382,9 @@ export const hostActionResolver = ({
           actions.push({
             title: <ActionTitle disabled={!enabled} description={reason} title="Change hostname" />,
             id: `button-edit-host-${hostname}`, // id is everchanging, not ideal for tests
-            onClick: () => onEditHost(host),
+            onClick: () => {
+              enabled && onEditHost(host);
+            },
             disabled: !enabled,
```

Comment:
we should use `isDisabled` instead of `disabled` and then we wont have to double check in `onClick` callback (applies for all others too)
```suggestion
            onClick: () => onEditHost(host),
            isDisabled: !enabled,
```

Reply:
done!